nemo/collections/asr/data/audio_to_text_dali.py

```diff
@@ -320,22 +324,69 @@ def __init__(
             self.pad_value = params['pad_value'] if 'pad_value' in params else 0.0
 
         with self.pipe:
-            audio, indices = dali.fn.readers.nemo_asr(
-                name="Reader",
-                manifest_filepaths=manifest_filepath.split(','),
-                dtype=dali.types.FLOAT,
-                downmix=True,
-                sample_rate=float(self.sample_rate),
-                min_duration=min_duration,
-                max_duration=max_duration,
-                read_sample_rate=False,
-                read_text=False,
-                read_idxs=True,
-                random_shuffle=shuffle,
-                shard_id=self.shard_id,
-                num_shards=self.num_shards,
-                pad_last_batch=True,
-            )
+            if audio_tar_filepaths is None and audio_tar_index_filepaths is None:
+                audio, indices = dali.fn.readers.nemo_asr(
+                    name="Reader",
+                    manifest_filepaths=manifest_filepath.split(','),
+                    dtype=dali.types.FLOAT,
+                    downmix=True,
+                    sample_rate=float(self.sample_rate),
+                    min_duration=min_duration,
+                    max_duration=max_duration,
+                    read_sample_rate=False,
+                    read_text=False,
+                    read_idxs=True,
+                    random_shuffle=shuffle,
+                    shard_id=self.shard_id,
+                    num_shards=self.num_shards,
+                    pad_last_batch=True,
+                )
+
+                self.is_tarred_dataset = False
+
+            elif audio_tar_filepaths is not None and audio_tar_index_filepaths is not None:
+                audio_tar_filepaths = expand_audio_filepaths(
+                    audio_tar_filepaths, shard_strategy=shard_strategy, world_size=world_size, global_rank=global_rank
+                )
+                audio_tar_index_filepaths = expand_audio_filepaths(
+                    audio_tar_index_filepaths,
+                    shard_strategy=shard_strategy,
+                    world_size=world_size,
+                    global_rank=global_rank,
+                )
+
+                if len(audio_tar_filepaths) != len(audio_tar_index_filepaths):
```

Comment:
```suggestion
                if len(audio_tar_filepaths) != len(audio_tar_index_filepaths) and len(audio_tar_index_filepaths) != 0:
```

Reply:
Accepted, patched manually as otherwise DCO complains.

Reply:
For the reason stated before it is not needed after all.